src/status_im/transport/message/v1/contact.cljs

```diff
@@ -0,0 +1,131 @@
+        (handlers/merge-fx cofx
+                           #_(protocol/ack message-id chat-id)
+                           (contacts/receive-contact-request-confirmation signature
+                                                                          this))))))
+
+(defrecord ContactUpdate [name profile-image]
+  message/StatusMessage
+  (send [this _ {:keys [db] :as cofx}]
+    (let [message-id (transport.utils/message-id this)
+          public-keys (remove nil? (map :public-key (vals (:contacts/contacts db))))]
+      (handlers/merge-fx cofx
+                         (protocol/multi-send-with-pubkey {:public-keys public-keys
+                                                           :payload     this}))))
+  (receive [this chat-id signature cofx]
+    (let [message-id (transport.utils/message-id this)]
+      (when (protocol/is-new? message-id)
+        (handlers/merge-fx cofx
+                           (contacts/receive-contact-update chat-id
+                                                            signature
+                                                            this))))))
+
+(handlers/register-handler-fx
+  ::send-new-sym-key
+  (fn [{:keys [db random-id] :as cofx} [_ {:keys [chat-id topic message sym-key sym-key-id]}]]
+    (let [{:keys [web3 current-public-key]} db]
+      (handlers/merge-fx cofx
+                         {:db (assoc-in db [:transport/chats chat-id :sym-key-id] sym-key-id)
+                          :shh/add-filter {:web3       web3
+                                           :sym-key-id sym-key-id
+                                           :topic      topic
+                                           :chat-id    chat-id}
+                          :data-store.transport/save {:chat-id chat-id
+                                                      :chat    (-> (get-in db [:transport/chats chat-id])
+                                                                   (assoc :sym-key-id sym-key-id)
+                                                                   ;;TODO (yenda) remove once go implements persistence
+                                                                   (assoc :sym-key sym-key))}}
+                         (message/send (NewContactKey. sym-key topic message)
+                                       chat-id)))))
+
+(handlers/register-handler-fx
+  ::add-new-sym-key
+  (fn [{:keys [db] :as cofx} [_ {:keys [sym-key-id sym-key chat-id topic message]}]]
+    (let [{:keys [web3 current-public-key]} db]
+      (handlers/merge-fx cofx
+                         {:db (assoc-in db [:transport/chats chat-id :sym-key-id] sym-key-id)
+                          :shh/add-filter {:web3       web3
+                                           :sym-key-id sym-key-id
+                                           :topic      topic
+                                           :chat-id    chat-id}
+                          :data-store.transport/save {:chat-id chat-id
```

Comment:
Not sure, but maybe this data store is an implementation detail that transport ns shouldn't care about? Not clear to me where this is "saving to", nor why I should care in the context of this namespace

Reply:
It's not clear to you where this is saving to because it is an implementation detail :) The fx implements it in the data-store namespaces, here you just say that you want to persist these informations because they need to be restored when restarting the app.